cmd/dep/status.go

```diff
@@ -335,6 +388,142 @@ func (cmd *statusCommand) validateFlags() error {
 	return nil
 }
 
+// OldStatus contains information about all the out of date packages in a project.
+type OldStatus struct {
+	ProjectRoot string
+	Constraint  gps.Constraint
+	Revision    gps.Revision
+	Latest      gps.Version
+}
+
+type rawOldStatus struct {
+	ProjectRoot, Constraint, Revision, Latest string
+}
+
+func (os OldStatus) getConsolidatedConstraint() string {
+	var constraint string
+	if os.Constraint != nil {
+		if v, ok := os.Constraint.(gps.Version); ok {
+			constraint = formatVersion(v)
+		} else {
+			constraint = os.Constraint.String()
+		}
+	}
+	return constraint
+}
+
+func (os OldStatus) getConsolidatedLatest(revSize uint8) string {
+	latest := ""
+	if os.Latest != nil {
+		switch revSize {
+		case shortRev:
+			latest = formatVersion(os.Latest)
+		case longRev:
+			latest = os.Latest.String()
+		}
+	}
+	return latest
+}
+
+func (os OldStatus) marshalJSON() *rawOldStatus {
+	return &rawOldStatus{
+		ProjectRoot: os.ProjectRoot,
+		Constraint:  os.getConsolidatedConstraint(),
+		Revision:    string(os.Revision),
+		Latest:      os.getConsolidatedLatest(longRev),
+	}
+}
+
+func (cmd *statusCommand) runOld(ctx *dep.Ctx, out oldOutputter, p *dep.Project, sm gps.SourceManager) error {
+	// While the network churns on ListVersions() requests, statically analyze
+	// code from the current project.
+	ptree, err := p.ParseRootPackageTree()
+	if err != nil {
+		return err
+	}
+
+	// Set up a solver in order to check the InputHash.
+	params := gps.SolveParameters{
+		ProjectAnalyzer: dep.Analyzer{},
+		RootDir:         p.AbsRoot,
+		RootPackageTree: ptree,
+		Manifest:        p.Manifest,
+		// Locks aren't a part of the input hash check, so we can omit it.
+	}
```

Comment:
Let's add support for log output when `-v` is passed, similar to https://github.com/golang/dep/blob/v0.4.1/cmd/dep/status.go#L445-L449 .
Since the solving takes some time, a verbose mode would be good to have to see what's happening.
Also, I think we should add a line telling the user that's it's solving and checking for updates. Right now, there's no message and it waits for long, which could be confusing to the user.